### ISSUE TITLE
Bug: App doesn't start, keeps loading

### ISSUE THREAD

**Issue by davidbergstrom** [bitgapp/eqMac#284]
**Disclaimer:** 
Please go through [existing issues](https://github.com/bitgapp/eqMac/issues?q=is%3Aopen+is%3Aissue+label%3A%22type%3A+bug%22) to avoid creating duplicates. If you found a similar issue please post a comment there. If you are uncertain your issue is related to others then create the issue. Obvious ignorance of this rule might get you blocked from posting new issues. Please tick the box below to show that you have followed the guideline.
- [x] I have checked for a similar issue and sure it hasn't been reported before.

**Describe the bug**
When I try to start the app I only get the booting screen with a spinner, and then I'm stuck there.

**Steps to Reproduce**
Steps to reproduce the behaviour (feel free to change the placeholder as you need):
1. Start the app
2. Error occurs
**Expected behaviour**
The app starts.

**Setup information:**
 - macOS Version: 10.15.4
- eqMac UI Version: Latest

**Comment by nodeful** [comment]
Did you by any chance block eqMac from accessing the internet?

**Comment by ils3000** [comment]
Same issue.
macOS: 10.13.6
Only started happening when updated the app to 0.2.0.
Made sure to check if eqMac has access to Internet in Mac's Privacy Settings, and it does have access.
Thanks!

**Comment by davidbergstrom** [comment]
> Did you by any chance block eqMac from accessing the internet?

I haven't changed anything in my settings, but I tried adding it to the list of allows in my firewall settings, and now it launches!

**Comment by mmitchellg5** [comment]
I'm hoping that its not an expected behavior that the app won't launch without access to the internet ...

**Comment by leeduc** [comment]
Yesterday It was working but today loading forever, only white :(

**Comment by nodeful** [comment]
Even though the UI of eqMac is hosted from a remote server, for over the air updates, it is shipped with a local version that it runs on a small server in case there is no internet access. How do you guys block eqMac's access to the internet?

**Comment by nodeful** [comment]
I have installed Little Snitch 4 and eqMac works for me even tho I blocked it's internet access

**Comment by nodeful** [comment]
![image](https://user-images.githubusercontent.com/8472525/83301488-aa9fa180-a1f1-11ea-82d1-915cbe31a111.png)

**Comment by trentjk** [comment]
This issue is also happening to me as well. I have tried deleting/uninstalling, re-installing, and restarting my computer in several combinations, and I still get a loading icon every time I try to open the app. I suspect it may be because at one point I force quit the app, and/or its processes, but I am not sure why that would cause it to stop working, especially after fresh re-install. I have added eqMac to my firewall, which did not work.

(I also recently installed sound flower, but I believe those two were working in conjunction for some time before the issue started happening).

**Comment by MilesConn** [comment]
I want to say I have a similar bug. Sometimes I open eqMac and it loads indefinitely other times it opens and then I don't see an interface at all despite it being a running process. I'm fairly certain this is related. I haven't blocked its internet or changed any settings besides my display preferences for it. 

I have also completely uninstalled the app using AppCleaner which removes plists and stuff and reinstalled it and it displays the same behaviour. 
```
ps -au $USER | grep eqMac                                                                  1 ↵
  501 27308 ??         0:00.27 /Applications/eqMac.app/Contents/MacOS/eqMac -psn_0_13356220
  501 28679 ttys008    0:00.00 grep --color=auto --exclude-dir=.bzr --exclude-dir=CVS --exclude-dir=.git --exclude-dir=.hg --exclude-dir=.svn eqMac
```

**Comment by trentjk** [comment]
My issue was randomly resolved after leaving a fresh install on my system overnight. I came to my computer after leaving it for a while (It was shut down over night, and auto boot up in the morning) and it asked if I wanted to install the drivers. I said yes, and it installed and started working, and this bug has not occurred since then. If someone reading this has this issue, maybe attempt to shut down your computer for a little, then boot up, instead of restart? (Tbh, to my knowledge I thought that would have resulted in the same exact outcome, but maybe I’m wrong).

**Comment by SamsTheGreatest** [comment]
I have previously been using eqMac2, which I deleted completely from my machine. When I was trying to open the new eqMac app, neither driver installation nor permission-asking windows would appear. So, I executed the following steps:

1. Delete every eqMac file in your system
2. Open .dmg and drag into the Applications folder required.
3. Before launching the app go to Terminal and execute the following, it installed the drivers for me: 

- cd /

- bin/bash

- sudo /Applications/eqMac.app/Contents/Resources/install_driver.sh


8. Still won't launch, so I open the app manually via Terminal: 

- sudo /Applications/eqMac.app/Contents/MacOS/eqMac

App asks for the microphone use permission, however, it won't appear in System Preferences. It seems to recognize connected devices but won't work with audio. 

I hope it helps. Used Catalina 10.15.4.

**Comment by xphoenyx** [comment]
@SamsTheGreatest your solution did a trick for me, thank you! Although launching the app from the launchpad still results in the same infinite-loading window.

**Comment by MilesConn** [comment]
@SamsTheGreatest  this worked for me as well.

**Comment by HWiese1980** [comment]
@SamsTheGreatest Confirmed working for me. At least the starting from terminal part. Haven't tried starting it from the Launchpad again yet.

**Comment by MilesConn** [comment]
Whenever my computer goes to sleep the process dies but interestingly enough I can still see eqMac in audio devices. I'm not sure if it's a related issue.

**Comment by leeduc** [comment]
@SamsTheGreatest solution is helpful. 

I think Mac 10.15 has problem, eqMac only works when runs with sudo.

**Comment by zergeborg** [comment]
I can see two very disturbing things here. An audio app needs internet (why???), and it only runs with sudo (why again???). These are both huge security red alerts in my reading at least. The app should not need either of those.

**Comment by nodeful** [comment]
@zergeborg as I mentioned before the app does not need internet access to work. The app needs sudo priveleges only to install a driver, there's no other alternative to eqMac that doesn't need a similar driver. No need to spread your ignorant security nazi views if you don't have basic understanding of how the app works.

**Comment by zergeborg** [comment]
@nodeful Ignorant security nazi views? WTF is your problem? People in this thread reported to you that the app doesn't even start without internet connection and it only loads when started with sudo from the command line and your response is when people start raising their concerns is calling them ignorant nazis? Then WTF are you? Ignorant bigot or what??? Good to know your stance.

**Comment by SamsTheGreatest** [comment]
In a while after a couple of restarts, it requested access to the microphone and started working without sudo.

**Comment by t0mn** [comment]
OK that worked - I installed eqMac and then added it to the firewall as allowed.  When I started the app it sat there for a while installing the audio driver, then started as normal.  So we appear to have a work-around, not sure why it needs a firewall rule though...

**Comment by jandekkerongit** [comment]
I also have this issue. None of the terminal related solutions worked for me. Opening the app shows a window with a spinner that never gets anywhere.

**Comment by Ashteroide** [comment]
Downloaded v0.3.1, didn't work until I restarted my mac, it doesn't like when sound is playing if you reset playback but it's working now

**Comment by nodeful** [comment]
Can this issue relating to the UI not loading be closed now?
In update v0.3.1 everything should work. Please update either through the app or redownload on the website or GitHub releases. Thank you

@Ashteroide this issue is related to the app you using to play sound deciding to pipe sound directly to your device and circumventing eqMac (for me happens a lot in Chrome and YouTube but usually a full restart of Chrome solves the issue)

**Comment by Ashteroide** [comment]
@nodeful I am trying to reinstall the drivers but nothing it working, I originally tried after constant failure to use the install_driver.sh in the app itself but that just removed all output devices on my Mac, although internal still works without it showing

**Comment by nodeful** [comment]
This issue should be resolved in [v0.3.1](https://github.com/bitgapp/eqMac/releases/tag/v0.3.1).
Please update the app and let me know if this issue is still relevant. Thank you
@Ashteroide restart your mac and if the issue still persists then report to #291

**Comment by t0mn** [comment]
Thank you, all good now.  Installed update, removed firewall rule & restarted app - all OK.  Thanks for quick response.

**Comment by Ashteroide** [comment]
@nodeful I removed the bitgapp folder in the Application Support and Webkit and the application itself, downloaded the v0.3.1 dmg again, ran it and put the app to the main macOS Application folder rather than the home Application and everything it working now, my Bass Boost preset it working, I didn't have to reinstall any drivers, must've been from before when I ran the install_driver.sh

**Comment by jandekkerongit** [comment]
This issue is not resolved. I tried multiple "solutions" today, always using a fresh download from the website.

**Comment by sakwads** [comment]
Hello Everyone, I had the issue with the loading window. Running Eqmac 0.3.0, Mac OS Catalina 10.15.5

1. First I killed the process in Activity Monitor. 
2. I went to finder and searched for Eqmac. I deleted everything that was associated with Eqmac (Including Eqmac2 files still lingering)
3. Downloaded EqMac again and installed. I accepted that it was downloaded from the internet (Apple Securit Policy) and it is now working again. I have quit and restarted and the interface now launches

I hope this helps
